### PR TITLE
Fix examples/mnist_net2net.py for Keras 2

### LINKED ISSUE #5690: mnist_net2net.py  run error
When I run mnist_net2net.py, it is  "AssertionError: weight and bias from same layer should have compatible shapes"

### LINKED ISSUE #6539: Incorrect weights' shape and questions about get_layer in example/mnist_net2net.py
In Keras 1: the shape of weight in Conv2D layer is of (filters,channels,height,weight), but in Keras 2: the shape is (kw,kh,num_channel,filters). I have fixed this bug, please ref `mnist_net2net_fixed.py` in gist:

https://gist.github.com/luzai/3336bd582650ff2c8c4ff97b1186afcd

Meanwhile, I find that `model.get_layer(name)` returns a `None` even if `model.layers` contain this layer. Please ref to `get_layer_bug.py` in gist.
Temporarily, I fix this bug by some tricks shown below, but may I ask why  `model.get_layer(name)` returns a `None`? 
```
if model.get_layer(name='conv2-deeper') is None:
	names2ind = {layer.name: ind for ind, layer in enumerate(model.layers)}
	ind = names2ind['conv2-deeper']
	model.layers[ind].set_weights(new_weights)
else:
	model.get_layer(name='conv2-deeper').set_weights(new_weights)
```

### PR DESCRIPTION
**Pull request by ozabluda**
examples/mnist_net2net.py was broken since transition to Keras-2, because filter layout changed, producing "AssertionError: weight and bias from same layer should have compatible shapes".

Also some of the validation losses had typos in them, made moot by the new validation losses, which are always stricter less than the old ones.

Resolves #5690, resolves #6539.

### CODE REVIEW

File: examples/mnist_net2net.py [hunk 1]
```diff
@@ -193,12 +198,12 @@ def deeper2net_conv2d(teacher_w):
 
     # Arguments
         teacher_w: `weight` of previous conv2d layer,
-          of shape (filters, num_channel, kh, kw)
+          of shape (kw, kh, num_channel, filters)
```
Comment (fchollet) [inline examples/mnist_net2net.py]: Shape is actually NHWC, i.e. height before width

File: examples/mnist_net2net.py [hunk 2]
```diff
@@ -193,12 +198,12 @@ def deeper2net_conv2d(teacher_w):
...
     '''
-    filters, num_channel, kh, kw = teacher_w.shape
-    student_w = np.zeros((filters, filters, kh, kw))
+    kw, kh, num_channel, filters = teacher_w.shape
```
Comment (fchollet) [inline examples/mnist_net2net.py]: Same here and below

File: examples/mnist_net2net.py (hunk 1 above)
Comment (ozabluda) [reply examples/mnist_net2net.py]: Fixed. It's filters, not images, so HWCK, right?

**Comment by ozabluda** [comment]
I tried set_image_data_format('channels_first') but it didn't work. Had it worked I wouldn't have done it the hard way :-)

File: examples/mnist_net2net.py (hunk 1 above)
Comment (fchollet) [reply examples/mnist_net2net.py]: Yes, that's correct

**Pull request merged**